mqtt/tls-connect-to-broker.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: default
+title: A short (<10 words) summary of the task addressed
+slug:
+  - label: mqtt
+    url: /#mqtt
+  - secure connect
+---
+
+### Problem
+
+You want to have an encrypted connection to your Mosquitto MQTT broker.
+
+### Solution
+
+Create a valid set of certificates and keys for the broker to use.
+
+Change the configuration of the broker to start a TLS encrypted port (`mqtts`) using the above.
+
+Alter the <code class="node">MQTT Config</code> node, changing the "Server" name to start with `mqtts://`.
+
+#### Example
+
+**Valid Certificate Creation**
+
+THe easiest approach for this is to use [Let's Encrypt](). This is beyond the scopy of this article but there are plenty of examples and tutorials available on the Internet. For this to work successfully, you also need to be able to use a registered domain name on your internal network because you cannot use Let's Encrypt with IP addresses or non-public domain names.
+
+Alternatively, you can create a self-signed set of certificates. Again, this is beyond the scope of the article. However, you may need to create a trusted root certificate and provide its public cert instead of the one that Debian provides that is listed in the configuration below.
+
+**Mosquitto configuration**
+
+This assumes that you are using Let's Encrypt or other certificates signed by a root CA already trusted by the Debian operating system.
+
+Note the entries in `<...>` which need to be replaced with your own folders and files.
+
+This goes into a file of any name of the form `*.config` in the folder `/etc/mosquitto/conf.d/`. So you have to edit it with root privalages (e.g. using `sudo`).
+
+{{ page.lcb }}% raw %}
+~~~text
+# Default Listener: 1883
+port 1883
+# Bind the default listener to localhost only if you want to force external connections to be TLS only
+#bind_address localhost
+
+# Secure listener
+listener 8883
+# TLS
+## This is standard and should always be this when using Let's Encrypt
+## If using a self-signed certificate, this needs to be your custom Root CA public certificate
+cafile /etc/ssl/certs/DST_Root_CA_X3.pem
+## These are from your installation of LE
+certfile /<path-to-LE-cert-files>/<fullchain>.cer
+keyfile /<path-to-LE-cert-files>/<private-key-name>.key
+## Forces use of modern version of TLS to avoid security issues
+tls_version tlsv1.2
+
+## Forces ALL CLIENTs to provide a valid certificate - change the node config to allow this from NR
+#require_certificate true
```

Comment:
`require_certificate` is an option for user authentication, not for encryption. Maybe better to completely leave it out here?

Reply:
A not unreasonable thought. I've included it because it causes confusion and leaving it out wouldn't remove that confusion. So I believe it should remain as is - though with the amended comment as per Paul's suggestion above.